app/components/custom/poll/answers/form_component.rb

```diff
@@ -0,0 +1,24 @@
+class Poll::Answers::FormComponent < ApplicationComponent
+  attr_accessor :answers, :poll, :token
+  delegate :current_user, to: :helpers
+
+  def initialize(poll, token, answers)
+    @poll = poll
+    @token = token
+    @answers = answers
+  end
+
+  def questions
+    poll.questions.order(:created_at)
+  end
+
+  def question_answer(question)
+    answers.find { |answer| answer.question == question }
+  end
```

Comment:
Possible simplification:

```diff
diff --git a/app/components/custom/poll/answers/form_component.html.erb b/app/components/custom/poll/answers/form_component.html.erb
index 08280b085..a9cdc633b 100644
--- a/app/components/custom/poll/answers/form_component.html.erb
+++ b/app/components/custom/poll/answers/form_component.html.erb
@@ -2,8 +2,8 @@
   <%= f.invisible_captcha :title %>
   <%= hidden_field_tag :token, token, class: "js-poll-answers-form-token" %>
   <%= render ErrorsComponent.new(error_message) %>
-  <% questions.each do |question| %>
-    <%= render Poll::Question::FieldsComponent.new(f, question, question_answer(question)) %>
+  <% answers.each do |answer| %>
+    <%= render Poll::Question::FieldsComponent.new(f, answer) %>
   <% end %>
 
   <%= f.submit(class: "button", value: t("polls.answers.form.submit_button"), disabled: poll.expired?) %>
diff --git a/app/components/custom/poll/answers/form_component.rb b/app/components/custom/poll/answers/form_component.rb
index b1c1a026f..b4662f6e2 100644
--- a/app/components/custom/poll/answers/form_component.rb
+++ b/app/components/custom/poll/answers/form_component.rb
@@ -8,14 +8,6 @@ class Poll::Answers::FormComponent < ApplicationComponent
     @answers = answers
   end
 
-  def questions
-    poll.questions.order(:created_at)
-  end
-
-  def question_answer(question)
-    answers.find { |answer| answer.question == question }
-  end
-
   def error_message
     count = answers.select { |a| a.errors.any? }.map(&:errors).flatten.count
 
diff --git a/app/components/custom/poll/question/fields_component.rb b/app/components/custom/poll/question/fields_component.rb
index ad58c1ce8..516409371 100644
--- a/app/components/custom/poll/question/fields_component.rb
+++ b/app/components/custom/poll/question/fields_component.rb
@@ -1,13 +1,16 @@
 class Poll::Question::FieldsComponent < ApplicationComponent
-  attr_accessor :answer, :form, :question
+  attr_accessor :answer, :form
   delegate :current_user, to: :helpers
 
-  def initialize(form, question, answer)
+  def initialize(form, answer)
     @form = form
-    @question = question
     @answer = answer
   end
 
+  def question
+    answer.question
+  end
+
   def errors_for(field)
     return unless answer.errors.include?(field)
```

Renaming the component to `Poll::Answers::FieldsComponent` could also be an option (after all, the fields refer to an answer, and not to a question :thinking:).